packet_ubuntu20-calico-aio job is failing

<!-- Please only use this template for submitting reports about failing tests in Kubespray CI jobs -->

**Which jobs are failing**:

packet_ubuntu20-calico-aio 

**Which test(s) are failing**:

packet_ubuntu20-calico-aio 

**Since when has it been failing**:

For awhile

**Testgrid link**:

Don't know what a testgrid link is, but here's a link to the failing job:

https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/-/jobs/3593920787

**Reason for failure**:

It appears the templating of Ansible files is somehow producing a corrupt file

**Anything else we need to know**:

No

Thanks for submitting this issue.
The log looks like
```
TASK [packet-ci : Template vm files for CI job] ********************************
task path: /builds/kargo-ci/kubernetes-sigs-kubespray/tests/cloud_playbooks/roles/packet-ci/tasks/create-vms.yml:13
changed: [localhost] => (item=1) => {"ansible_index_var": "vm_id", "ansible_loop_var": "item", "changed": true, "checksum": "c0dcd6b1c137e2465b0bf6b662908d820db899b5", "dest": "/tmp/745716983-3595324634/instance-0.yml", "gid": 0, "group": "root", "item": 1, "md5sum": "0dbb4dbd5bb60efbb51f8cb397e9635c", "mode": "0644", "owner": "root", "size": 1956, "src": "/root/.ansible/tmp/ansible-tmp-1673545171.4103062-141-132224033852724/source", "state": "file", "uid": 0, "vm_id": 0}
TASK [packet-ci : Start vms for CI job] ****************************************
task path: /builds/kargo-ci/kubernetes-sigs-kubespray/tests/cloud_playbooks/roles/packet-ci/tasks/create-vms.yml:22
failed: [localhost] (item=1) => {"ansible_index_var": "vm_id", "ansible_loop_var": "item", "changed": false, "cmd": ["kubectl", "apply", "-f", "/tmp/745716983-3595324634/instance-0.yml"], "delta": "0:00:00.268522", "end": "2023-01-12 17:39:32.291793", "item": 1, "msg": "non-zero return code", "rc": 1, "start": "2023-01-12 17:39:32.023271", "stderr": "error: proto: cannot parse invalid wire-format data", "stderr_lines": ["error: proto: cannot parse invalid wire-format data"], "stdout": "", "stdout_lines": [], "vm_id": 0}
PLAY RECAP *********************************************************************
localhost                  : ok=5    changed=2    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
make: *** [Makefile:63: create-packet] Error 2
```

--- a/pipeline.Dockerfile
+++ b/pipeline.Dockerfile
@@ -45,3 +45,7 @@ RUN wget https://releases.hashicorp.com/vagrant/${VAGRANT_VERSION}/vagrant_${VAG
  dpkg -i vagrant_${VAGRANT_VERSION}_x86_64.deb && \
  rm vagrant_${VAGRANT_VERSION}_x86_64.deb && \
  vagrant plugin install vagrant-libvirt
+
+# Install Kubernetes collections
+RUN pip3 install kubernetes \
+    && ansible-galaxy collection install kubernetes.core

--- a/tests/cloud_playbooks/roles/packet-ci/tasks/create-vms.yml
+++ b/tests/cloud_playbooks/roles/packet-ci/tasks/create-vms.yml
@@ -11,20 +11,19 @@
     mode: 0755
 
 - name: Template vm files for CI job
-  template:
-    src: "vm.yml.j2"
-    dest: "/tmp/{{ test_name }}/instance-{{ vm_id }}.yml"
-    mode: 0644
+  set_fact:
+    vms_files: "{{ vms_files }} + [{{ lookup('ansible.builtin.template', 'vm.yml.j2') | from_yaml }}]"
+  vars:
+    vms_files: []
   loop: "{{ range(1, vm_count|int + 1, 1) | list }}"
   loop_control:
     index_var: vm_id
 
 - name: Start vms for CI job
-  command: "kubectl apply -f /tmp/{{ test_name }}/instance-{{ vm_id }}.yml"
+  kubernetes.core.k8s:
+    definition: "{{ item }}"
   changed_when: false
-  loop: "{{ range(1, vm_count|int + 1, 1) | list }}"
-  loop_control:
-    index_var: vm_id
+  loop: "{{ vms_files }}"
 
 - name: Wait for vms to have ipaddress assigned
   shell: "set -o pipefail && kubectl get vmis -n {{ test_name }} instance-{{ vm_id }} -o json | jq '.status.interfaces[].ipAddress' | tr -d '\"'"

